src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -273,6 +273,15 @@ private async Task<OpenApiResponse> GetResponseAsync(ApiDescription apiDescripti
         return parameters;
     }
 
+    private static bool IsRequired(ApiParameterDescription parameter)
+    {
+        var hasRequiredAttribute = parameter.ParameterDescriptor is IParameterInfoParameterDescriptor parameterInfoDescriptor &&
+            parameterInfoDescriptor.ParameterInfo.GetCustomAttributes(inherit: true).Any(attr => attr is RequiredAttribute);
+        // Per the OpenAPI specification, parameters that are sourced from the path
+        // are always required, regardless of the requiredness status of the parameter.
+        return parameter.Source == BindingSource.Path || parameter.IsRequired || hasRequiredAttribute;
```

Comment:
Is it worth refactoring to not run the code to find the attribute if the first two conditions are true?

Reply:
🤷🏽‍♀️ I think the check is fairly cheap and I like that the current ordering is priority-based (e.g. makes it clear that a parameter being sourced from the path will always short-circuit the remaining checks).